### PR TITLE
IO-811: Files.walk() direct and indirect callers fail to close the re…

### PR DESCRIPTION
**Pull request by labkey-adam**
…turned Stream<Path>

https://issues.apache.org/jira/browse/IO-811

### CODE REVIEW

File: src/main/java/org/apache/commons/io/FileUtils.java [hunk 1]
```diff
@@ -1996,7 +1996,28 @@ public static Iterator<File> iterateFiles(final File directory, final IOFileFilt
...
+            Stream<File> stream = streamFiles(d, recursive, extensions);
+            Iterator<File> iter = stream.iterator();
+
+            // Wrap the iterator to allow closing the stream after consumption
+            return new Iterator<File>(){
+                @Override
+                public boolean hasNext() {
+                    boolean ret = iter.hasNext();
+
+                    if (!ret)
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/io/FileUtils.java:2009]: Always use blocks -> { }

File: src/main/java/org/apache/commons/io/FileUtils.java [hunk 2]
```diff
@@ -1996,7 +1996,28 @@ public static Iterator<File> iterateFiles(final File directory, final IOFileFilt
      * @since 1.2
      */
     public static Iterator<File> iterateFiles(final File directory, final String[] extensions, final boolean recursive) {
-        return Uncheck.apply(d -> streamFiles(d, recursive, extensions).iterator(), directory);
+        return Uncheck.apply(d -> {
+            Stream<File> stream = streamFiles(d, recursive, extensions);
+            Iterator<File> iter = stream.iterator();
+
+            // Wrap the iterator to allow closing the stream after consumption
+            return new Iterator<File>(){
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/io/FileUtils.java:2004]: Missing space.

**Review by garydgregory — CHANGES_REQUESTED** [review]
Hello @labkey-adam 
Thank you for your PR. I'll have to debug later but in the meantime, note the formatting nits.

File: src/main/java/org/apache/commons/io/FileUtils.java (hunk 1 above)
Comment (labkey-adam) [reply src/main/java/org/apache/commons/io/FileUtils.java]: Done

File: src/main/java/org/apache/commons/io/FileUtils.java (hunk 2 above)
Comment (labkey-adam) [reply src/main/java/org/apache/commons/io/FileUtils.java]: Done

**Review by garydgregory — COMMENTED** [review]
Hello @labkey-adam 
Thank you for your PR. I've addressed these issues differently and credited you in `changes.xml`.

**Comment by garydgregory** [comment]
Closing, see above. Jira ticket resolved.

**Pull request closed without merging**